libavcodec/vaapi_encode_h265.c

```diff
@@ -299,11 +299,14 @@ static int vaapi_encode_h265_init_sequence_params(AVCodecContext *avctx)
     ptl->general_profile_idc   = avctx->profile;
     ptl->general_tier_flag     = priv->tier;
 
-    if (chroma_format == 1) {
-        ptl->general_profile_compatibility_flag[1] = bit_depth ==  8;
-        ptl->general_profile_compatibility_flag[2] = bit_depth <= 10;
+    ptl->general_profile_compatibility_flag[ptl->general_profile_idc] = 1;
+
+    if (ptl->general_profile_compatibility_flag[1])
```

Comment:
is it ptl->general_profile_compatibility_flag[2] in the condition ?

Reply:
Should be general_profile_compatibility_flag[1]. Here is spec statement:

Conformance of a bitstream to the Main profile is indicated by general_profile_idc being equal to 1 or
general_profile_compatibility_flag[ 1 ] being equal to 1. Conformance of a sub-layer representation with TemporalId
equal to i to the Main profile is indicated by sub_layer_profile_idc[ i ] being equal to 1 or
sub_layer_profile_compatibility_flag[ i ][ 1 ] being equal to 1.
NOTE – When general_profile_compatibility_flag[ 1 ] is equal to 1, general_profile_compatibility_flag[ 2 ] should also be equal
to 1. When sub_layer_profile_compatibility_flag[ i ][ 1 ] is equal to 1 for a value of i, sub_layer_profile_compatibility_flag[ i ][ 2 ]
should also be equal to 1.